demo/index.html

```diff
@@ -4,9 +4,9 @@
     <meta charset="utf-8" />
     <meta content="width=device-width,initial-scale=1.0" name="viewport" />
 
-    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css" rel="stylesheet">
+    <link href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css" rel="stylesheet">

```

Comment:
Yes, it does hurt because if somebody uses https, then they cannot request resources via http. If you want, you can force https but don't force http.